docking/pose_dock.py

```diff
@@ -15,130 +16,183 @@
 from fe import system
 from fe.utils import to_md_units
 
-
 from matplotlib import pyplot as plt
 
 
-# from timemachine.lib import LangevinIntegrator
-
-
-if __name__ == "__main__":
-
-    suppl = Chem.SDMolSupplier("tests/data/ligands_40.sdf", removeHs=False)
+def pose_dock(
+    guests_sdfile,
+    host_pdbfile,
+    outdir,
+    n_steps,
+    lowering_steps,
+    start_percent,
+    random_rotation=False,
+    constant_atoms=[],
+):
+    """Poses guests into a host by running short simulations in which the guests phase in over time
+
+    Parameters
+    ----------
+
+    guests_sdfile: path to input sdf with guests to pose/dock
+    host_pdbfile: path to host pdb file to dock into
+    outdir: where to write output (will be created if it does not already exist)
+    n_steps: how many total steps of simulation to do (recommended: 20000)
+    lowering_steps: how many steps to lower the guest over (recommended: 10000)
```

Comment:
10000 might be a bit too aggressive for a LangevinIntegrator with a default friction of 1. I was seeing the ligand insert itself into all kinds of weird geometries - but I think this is probably target/ligand specific.